src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java

```diff
@@ -0,0 +1,172 @@
+                return "INT";
+            case INT64:
+                return "BIGINT";
+            case FLOAT32:
+                return "FLOAT";
+            case FLOAT64:
+                return "DOUBLE";
+            case BOOLEAN:
+                return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
+            default:
+                return "VARCHAR(255)";
+        }
+    }
+
+    public String getIdColumnDefinition() {
+        try {
+            if (isPostgreSQL()) {
+                if (getPostgresMajorVersion() <= 9) {
+                    return "id SERIAL PRIMARY KEY";
+                } else {
+                    return "id INTEGER PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY";
+                }
+            } else if (isDB2()) {
+                return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+            } else if (isMySQL()) {
+                return "id INTEGER PRIMARY KEY AUTO_INCREMENT";
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+    }
+
+    public int getPostgresMajorVersion() {
+        Connection connection;
+        try {
+            connection = dataSource.getConnection();
+            DatabaseMetaData metaData = connection.getMetaData();
+            String databaseProductName = metaData.getDatabaseProductName();
+            if ("PostgreSQL".equals(databaseProductName)) {
+                int majorVersion = metaData.getDatabaseMajorVersion();
+                return majorVersion;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    /**
+     * Get ID column definition
+     *
+     * @param jdbcWriter
+     */
+    public String getIdColumnDefinition(JDBCWriter jdbcWriter) {
+        try {
+            if (isPostgreSQL()) {
+                if (getPostgresMajorVersion() <= 9) {
+                    return "id SERIAL PRIMARY KEY";
+                } else {
+                    return "id INTEGER PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY";
+                }
+            } else if (isDB2()) {
+                return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+            } else if (isMySQL()) {
+                return "id INTEGER PRIMARY KEY AUTO_INCREMENT";
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+    }
+
+    public String buildInsertStatement(String tableName, List<String> fieldNames) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("INSERT INTO ").append(tableName);
+        sb.append("(").append(String.join(", ", fieldNames)).append(")");
+        sb.append(" VALUES ");
+        sb.append("(").append(String.join(", ", Collections.nCopies(fieldNames.size(), "?"))).append(")");
```

Comment:
Do we have any command builder tests? It is not clear why we are adding an n fold collection of "?" to this command some tests would make this class easier to understand

Reply:
The preparestatement replaces those ? with values for example.

```java
PreparedStatement pstmt = con.prepareStatement("UPDATE EMPLOYEES
                                     SET SALARY = ? WHERE ID = ?");
   pstmt.setBigDecimal(1, 153833.00)
   pstmt.setInt(2, 110592)
```